File: sampleapp_ui/src/services/CommentService.js
import http from "../http-common";

class CommentService {
  getAllComments(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/comment/comments`, searchDTO);
  }

  get(commentId) {
    return this.getRequest(`/comment/${commentId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/comment?field=${matchData}`, null);
  }

  addComment(data) {
    return http.post("/comment/addComment", data);
  }

  update(data) {
  	return http.post("/comment/updateComment", data);
  }
  
  uploadImage(data,commentId) {
  	return http.postForm("/comment/uploadImage/"+commentId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CommentService();

File: src/main/java/com/sampleapp/service/CommentService.java
package com.sampleapp.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.sampleapp.domain.Comment;
import com.sampleapp.dto.CommentDTO;
import com.sampleapp.dto.CommentSearchDTO;
import com.sampleapp.dto.CommentPageDTO;
import com.sampleapp.dto.CommentConvertCriteriaDTO;
import com.sampleapp.service.GenericService;
import com.sampleapp.dto.common.RequestDTO;
import com.sampleapp.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CommentService extends GenericService<Comment, Integer> {

	List<Comment> findAll();

	ResultDTO addComment(CommentDTO commentDTO, RequestDTO requestDTO);

	ResultDTO updateComment(CommentDTO commentDTO, RequestDTO requestDTO);

    Page<Comment> getAllComments(Pageable pageable);

    Page<Comment> getAllComments(Specification<Comment> spec, Pageable pageable);

	ResponseEntity<CommentPageDTO> getComments(CommentSearchDTO commentSearchDTO);
	
	List<CommentDTO> convertCommentsToCommentDTOs(List<Comment> comments, CommentConvertCriteriaDTO convertCriteria);

	CommentDTO getCommentDTOById(Integer commentId);







}





